src/saucelabs/job.js

```diff
@@ -1,138 +1,149 @@
-import Promise from 'pinkie';
+import wd from 'wd';
 import SaucelabsRequestAdapter from './request';
-import wait from '../utils/wait';
 
-const CHECK_RESULTS_TIMEOUT = 1000 * 20;
-const MAX_RESTART_COUNT     = 3;
+
+const MAX_RESTART_COUNT        = 3;
+const BROWSER_INIT_RETRY_DELAY = 30 * 1000;
+const BROWSER_INIT_RETRIES     = 3;
+const BROWSER_INIT_DURATION    = 9 * 60 * 1000;
+
+
+wd.configureHttp({
+    retryDelay: BROWSER_INIT_RETRY_DELAY,
+    retries:    BROWSER_INIT_RETRIES,
+    timeout:    BROWSER_INIT_DURATION
+});
 
 
 //Job
 export default class Job {
     constructor (options, platform) {
         this.options = {
-            username:         options.username,
-            accessKey:        options.accessKey,
-            build:            options.build,
-            testName:         options.name,
-            tags:             options.tags,
-            urls:             options.urls,
-            timeout:          options.timeout,
-            tunnelIdentifier: options.tunnelIdentifier
+            username:            options.username,
+            accessKey:           options.accessKey,
+            build:               options.build,
+            testName:            options.testName,
+            tags:                options.tags,
+            urls:                options.urls,
+            tunnelIdentifier:    options.tunnelIdentifier,
+            checkResultsTimeout: options.timeout * 1000
         };
 
         this.requestAdapter = new SaucelabsRequestAdapter(this.options.username, this.options.accessKey);
         this.platform       = platform;
+        this.browser        = wd.promiseRemote('ondemand.saucelabs.com', 80, options.username, options.accessKey);
 
-        this.taskId       = null;
         this.status       = Job.STATUSES.INITIALIZED;
         this.restartCount = 0;
     }
 
     static STATUSES = {
         INITIALIZED: 'initialized',
-        QUEUED:      'queued',
         IN_PROGRESS: 'in progress',
         COMPLETED:   'completed',
         FAILED:      'failed'
     };
 
-    _waitForComplete () {
-        var job               = this;
-        var statusRequestData = {
-            'js tests': [this.taskId]
-        };
+    async _waitForTestResults (js, testsTimeout, start) {
+        if (!start)
+            start = new Date();
+
+        if (new Date() - start > testsTimeout)
+            throw new Error(`Timeout: test results doesn't found`);
+
+        var jsValue = await this.browser.eval(js);
```

Comment:
I don't know, what name is better for object, which returns from invoked dynamically expression.